slsim/Sources/QuasarCatalog/number_density_quasar.py

```diff
@@ -0,0 +1,224 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.interpolate import interp1d
+from astropy.table import Table
+
+""" 
+References: 
+Richards et al. 2005
+Richards et al. 2006
+Oguri & Marshall (2010)
+"""
+
+
+def M_star(z_value, h=0.72, zeta=2.98, xi=4.05, z_star=1.60):
+    """Calculates the break absolute magnitude of quasars for a given redshift according
+    to Eq. (11) in Oguri & Marshall (2010): DOI: 10.1111/j.1365-2966.2010.16639.x.
+
+    Parameters:
+    :param z_value: Redshift value.
+    :type z_value: float or np.ndarray
+    :param h: Hubble constant parameter H0/100, where H0 = 72 km s^-1 Mpc^-1.
+    :param zeta: (1) Best fit value of the observed evolution of the quasar luminosity function from SDSS DR3 survery (Richards et al. 2006: DOI: 10.1086/503559)
+    :param xi: (2) Best fit value of the observed evolution of the quasar luminosity function from SDSS DR3 survery (Richards et al. 2006: DOI: 10.1086/503559)
+    :param z_star: (3) Best fit value of the observed evolution of the quasar luminosity function from SDSS DR3 survery (Richards et al. 2006: DOI: 10.1086/503559)
+    :Note: You can adjust these values if needed
+
+    Returns:
+    :rtype: float or np.ndarray
+    :return: M_star value.
+    :unit: magnitudes (mag)
+    """
+    # Convert z_value to a numpy array if it isn't already
+    z_value = np.atleast_1d(z_value)
+
+    denominator = (np.sqrt(np.exp(xi * z_value)) + np.sqrt(np.exp(xi * z_star))) ** 2
+    result = (
+        -20.90
+        + (5 * np.log10(h))
+        - (
+            2.5
+            * np.log10(np.exp(zeta * z_value) * (1 + np.exp(xi * z_star)) / denominator)
+        )
+    )
+
+    # Handle zero denominator cases
+    result[denominator == 0] = np.nan
```

Comment:
Do we need this line? Because if denominator is zero, it indicates something is wrong and we need error message in this case. But, this line will continue the code and we will never catch the error.